test/chisel.py

```diff
@@ -0,0 +1,178 @@
+        for authz in error.updated:
+            updated_authz = json.loads(urllib2.urlopen(authz.uri).read())
+            domain = authz.body.identifier.value,
+            for c in updated_authz['challenges']:
+                if 'error' in c:
+                    err = c['error']
+                    raise ValidationError(domain, err['type'], err['detail'])
+        # If none of the authz's had an error, just re-raise.
+        raise
+    if cert_output != None:
+        pem = OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM,
+                                              cert_resource.body)
+        with open(cert_output, 'w') as f:
+            f.write(pem)
+    return cert_resource
+
+def http_01_answer(client, chall_body):
+    """Return an HTTP01Resource to server in response to the given challenge."""
+    response, validation = chall_body.response_and_validation(client.key)
+    return standalone.HTTP01RequestHandler.HTTP01Resource(
+          chall=chall_body.chall, response=response,
+          validation=validation)
+
+def auth_and_issue(domains, email=None, cert_output=None, client=None):
+    """Make authzs for each of the given domains, set up a server to answer the
+       challenges in those authzs, tell the ACME server to validate the challenges,
+       then poll for the authzs to be ready and issue a cert."""
+    if client == None:
+        client = make_client(email)
+    authzs, challenges = make_authzs(client, domains)
+    port = 5002
+    answers = set([http_01_answer(client, c) for c in challenges])
+    server = standalone.HTTP01Server(("", port), answers)
+    thread = threading.Thread(target=server.serve_forever)
+    thread.start()
+
+    # Loop until the HTTP01Server is ready.
+    while True:
+        try:
+            urllib2.urlopen("http://localhost:%d" % port)
+            break
+        except urllib2.URLError:
+            time.sleep(0.1)
+
+    try:
+        for chall_body in challenges:
+            client.answer_challenge(chall_body, chall_body.response(client.key))
+        cert_resource = issue(client, authzs, cert_output)
+        return cert_resource
+    finally:
+        server.shutdown()
+        server.server_close()
+        thread.join()
+
+def expect_problem(problem_type, func):
+    """Run a function. If it raises a ValidationError or messages.Error that
+       contains the given problem_type, return. If it raises no error or the wrong
+       error, raise an exception."""
+    ok = False
+    try:
+        func()
+    except ValidationError as e:
+        if e.problem_type == problem_type:
+            ok = True
+        else:
+            raise
+    except messages.Error as e:
+        if problem_type in e.__str__():
+            ok = True
+        else:
+            raise
+    if not ok:
+        raise Exception("Expected %s, got no error" % problem_type)
+
+if __name__ == "__main__":
+    try:
+        auth_and_issue(sys.argv[1:])
```

Comment:
Can you catch the case where `len(sys.argv) == 1` and print an appropriate usage message?